packages/react-slider/src/components/RangedSlider/RangedSlider.test.tsx

```diff
@@ -0,0 +1,562 @@
+    expect(upperInputRef.current?.value).toBe('20');
+  });
+
+  it('handles a negative step prop', () => {
+    const lowerInputRef = React.createRef<HTMLInputElement>();
+    const upperInputRef = React.createRef<HTMLInputElement>();
+    const onChange = jest.fn();
+
+    render(
+      <RangedSlider
+        defaultValue={[20, 20]}
+        step={-3}
+        onChange={onChange}
+        inputLower={{ ref: lowerInputRef }}
+        inputUpper={{ ref: upperInputRef }}
+      />,
+    );
+
+    fireEvent.keyDown(lowerInputRef.current!, { key: 'ArrowUp' });
+    expect(onChange.mock.calls[0][1]).toEqual({ value: [17, 20] });
+    expect(lowerInputRef.current?.value).toBe('17');
+    expect(upperInputRef.current?.value).toBe('20');
+
+    fireEvent.keyDown(upperInputRef.current!, { key: 'ArrowUp' });
+    fireEvent.keyDown(upperInputRef.current!, { key: 'ArrowUp' });
+    expect(onChange.mock.calls[2][1]).toEqual({ value: [14, 17] });
+    expect(lowerInputRef.current?.value).toBe('14');
+    expect(upperInputRef.current?.value).toBe('17');
+  });
+
+  it('handles a decimal step prop', () => {
+    const lowerInputRef = React.createRef<HTMLInputElement>();
+    const upperInputRef = React.createRef<HTMLInputElement>();
+    const onChange = jest.fn();
+
+    render(
+      <RangedSlider
+        defaultValue={[20, 20]}
+        step={0.0001}
+        onChange={onChange}
+        inputLower={{ ref: lowerInputRef }}
+        inputUpper={{ ref: upperInputRef }}
+      />,
+    );
+
+    fireEvent.keyDown(upperInputRef.current!, { key: 'ArrowUp' });
+    expect(onChange.mock.calls[0][1]).toEqual({ value: [20, 20.0001] });
+    expect(lowerInputRef.current?.value).toBe('20');
+    expect(upperInputRef.current?.value).toBe('20.0001');
+
+    fireEvent.keyDown(upperInputRef.current!, { key: 'ArrowDown' });
+    fireEvent.keyDown(upperInputRef.current!, { key: 'ArrowDown' });
+    expect(onChange.mock.calls[2][1]).toEqual({ value: [19.9999, 20] });
+    expect(lowerInputRef.current?.value).toBe('19.9999');
+    expect(upperInputRef.current?.value).toBe('20');
+  });
+
+  it('applies focus to inputLower', () => {
+    const inputRef = React.createRef<HTMLInputElement>();
+    render(<RangedSlider inputLower={{ ref: inputRef }} />);
+    inputRef?.current?.focus();
+    expect(document.activeElement).toEqual(inputRef.current);
+  });
+
+  it('applies focus to inputUpper', () => {
+    const inputRef = React.createRef<HTMLInputElement>();
+    render(<RangedSlider inputUpper={{ ref: inputRef }} />);
+    inputRef?.current?.focus();
+    expect(document.activeElement).toEqual(inputRef.current);
+  });
+
+  it('does not allow focus on disabled RangedSlider', () => {
+    const sliderRef = React.createRef<HTMLInputElement>();
+
+    render(<RangedSlider ref={sliderRef} disabled />);
+
+    expect(document.activeElement).toEqual(document.body);
+    sliderRef?.current?.focus();
+    expect(document.activeElement).toEqual(document.body);
```

Comment:
We should add trying to focus on the inputs too since they should also be disabled by passing the `disabled` prop.